### PR TITLE
🐛 slash command examples

### PR DESCRIPTION
**Pull request by Nico105**
_I haven't really used slash commands yet, so don't roast me if smth is totally wrong 😆_

- should we make all `.catch` empheral ?

- Is it fine to write the `data` out?
Since we could remove that and for registering just do `delete command.run`.
= "save 1 object"
But that might confuse some people.

- Is `interaction.client.giveawaysManager` fine?
or should we provide the client as the 2nd command argument? since the above format could seem rather long.

### CODE REVIEW

**Review by Dragonizedpizza — CHANGES_REQUESTED** [review]
I think the snake case in the reroll command is unnecessary, considering that this is for the users to view, and Id looks weird. Also, the options' names must be lowercase (Discord API requires so), and I feel like the arrow function `run: async (interaction) => {` is unnecessary, when you could just do `async run(interaction) {`

**Comment by Nico105** [comment]
> I think the snake case in the reroll command is unnecessary, considering that this is for the users to view, and Id looks weird. Also, the options' names must be lowercase (Discord API requires so), and I feel like the arrow function `run: async (interaction) => {` is unnecessary, when you could just do `async run(interaction) {`

Now better?

**Comment by Dragonizedpizza** [comment]
> > I think the snake case in the reroll command is unnecessary, considering that this is for the users to view, and Id looks weird. Also, the options' names must be lowercase (Discord API requires so), and I feel like the arrow function `run: async (interaction) => {` is unnecessary, when you could just do `async run(interaction) {`
> 
> Now better?

Yes, but why is the command run function async even though you're using .then?

**Comment by Dragonizedpizza** [comment]
use async/await, its cleaner

**Comment by Dragonizedpizza** [comment]
```js
await client.giveawaysManager.start(interaction.channel, { /** */ }),catch(() => { /** */ });
interaction.reply(yes)
```

**Comment by Nico105** [comment]
> ```js
> await client.giveawaysManager.start(interaction.channel, { /** */ }),catch(() => { /** */ });
> interaction.reply("yes")
> ```
> 
> also, I think the replying should be deferred incase for whatever reason it takes more than 3 seconds.

applied, tho i went with try catch cause returning would have been a little "fancy" otherwise

File: src/Giveaway.js [hunk 1]
```diff
@@ -543,7 +543,7 @@ class Giveaway extends EventEmitter {
                         this.messages.winMessage
                             .substr(0, this.messages.winMessage.indexOf('{winners}'))
                             .replace('{prize}', this.prize)
-                            .replace('{messageURL}', this.messageURL)
+                            .replace('{messageURL}', this.messageURL),
```
Comment (Nico105) [inline src/Giveaway.js:546]:
```suggestion
                            .replace('{messageURL}', this.messageURL)
```

**Review by Androz2091 — COMMENTED** [review]
Thank you for this very huge PR. I guess that the best way is to keep minimal examples (I will make a commit don't worry), but the best would be to create an example bot like the one in the examples folder of discord-player. I will make it too, based on this PR. Thank you very much again, sorry this close this.

**Pull request closed without merging**